src/main/java/Main.java

```diff
@@ -50,56 +58,77 @@ private static BattleshipModel getModelFromReq(Request req){
     //}
     //For testing purposes
 
-    private static Boolean placeShip(Request req, Response res) {
+    private static String placeShip(Request req) {
         //collect info to put this ship into the model
         BattleshipModel model = game.getModel();
         //get the name
-        String name = req.attribute("name");
+        String name = req.params("id");
+
         //get the start
-        Coordinate start = new Coordinate(req.attribute("Across"), req.attribute("Down"));
+        Coordinate start = new Coordinate(Integer.parseInt(req.params("col")), Integer.parseInt(req.params("row")));
+
         //get orientation
-        String orientation = req.attribute("orientation");
+        String orientation = req.params("orientation");
+
         //set the scan boolean variable
-        boolean canScan = (name.equals("submarine"));
+        boolean canScan = true;
+
         //get length
-        int length = req.attribute("size");
+        int length=0;
+        if(name.equals("aircraftCarrier")){
```

Comment:
This is a lot of repeated code, but that's kind of necessary here. It might be cleaner to make a map, and a single if statement for the canScan of the submarine. This is fine, anyways.